pilot/proxy/envoy/policy.go

```diff
@@ -120,3 +125,128 @@ func applyClusterPolicy(cluster *Cluster,
 		}
 	}
 }
+
+func applyLoadBalancePolicy(cluster *Cluster, policy *routingv2.LoadBalancerSettings) {
+	if policy == nil || cluster.Type == ClusterTypeOriginalDST {
+		return
+	}
+
+	if consistent := policy.GetConsistentHash(); consistent != nil {
+		// cluster.LbType = LbTypeRingHash
+		// TODO: need to set special values in the route config
+		// see: https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/load_balancing.html#ring-hash
+		log.Warn("Consistent hash load balancing type is not currently supported")
+	} else {
+		switch policy.GetSimple() {
+		case routingv2.LoadBalancerSettings_LEAST_CONN:
+			cluster.LbType = LbTypeLeastRequest
+		case routingv2.LoadBalancerSettings_RANDOM:
+			cluster.LbType = LbTypeRandom
+		case routingv2.LoadBalancerSettings_PASSTHROUGH:
+			cluster.LbType = LbTypeOriginalDST // FIXME: double-check
+		default:
+			cluster.LbType = LbTypeRoundRobin
+		}
+	}
+}
+
+func buildOutlierDetection(outlier *routingv2.OutlierDetection) *OutlierDetection {
+	if outlier != nil && outlier.Http != nil {
+		out := &OutlierDetection{
+			ConsecutiveErrors:  5,
+			IntervalMS:         10000, // 10s
+			BaseEjectionTimeMS: 30000, // 30s
```

Comment:
here and above, `int64(30 * time.Second)`, `int64(10 * time.Second)`

Reply:
That won't work. `time.Duration` is nanoseconds:
```golang
const (
	Nanosecond  Duration = 1
	Microsecond          = 1000 * Nanosecond
	Millisecond          = 1000 * Microsecond
	Second               = 1000 * Millisecond
	Minute               = 60 * Second
	Hour                 = 60 * Minute
)```